subcmd/edit.go

```diff
@@ -0,0 +1,218 @@
+
+	// Build opt dir
+	if hasChanges {
+		err = builder.Build(false)
+		if err != nil {
+			return &Error{Code: 12, Msg: "Could not build " + pathutil.VimVoltDir() + ": " + err.Error()}
+		}
+	}
+
+	return nil
+}
+
+func (cmd *editCmd) doEdit(reposPathList []pathutil.ReposPath) (bool, error) {
+	// Read lock.json
+	lockJSON, err := lockjson.Read()
+	if err != nil {
+		return false, err
+	}
+
+	// Read config.toml
+	cfg, err := config.Read()
+	if err != nil {
+		return false, errors.New("could not read config.toml: " + err.Error())
+	}
+
+	viitor, err := cmd.identifyEditor(cfg)
+	if err != nil || viitor == "" {
+		//FIXME: Which error code to use?
+		return false, &Error{Code: 30, Msg: "No usable viitor found"}
+	}
+
+	changeWasMade := false
+	//FIXME: Run single vim instance? Leads to problems if the configured
+	//editor does not support to open multiple files
+	for _, reposPath := range reposPathList {
+
+		// Edit plugconf file
+		plugconfPath := reposPath.Plugconf()
+
+		// Install a new template if none exists
+		if !pathutil.Exists(plugconfPath) {
+			getCmd := new(getCmd)
+			logger.Debugf("Installing new plugconf for '%s'.", reposPath)
+			getCmd.downloadPlugconf(reposPath)
+		}
+
+		// Remember modification time before opening the editor
+		info, err := os.Stat(plugconfPath)
+		if err != nil {
+			return false, err
+		}
+		mTimeBefore := info.ModTime()
+
+		// Call the editor with the plugconf file
+		vimCmd := exec.Command(viitor, plugconfPath)
+		vimCmd.Stdin = os.Stdin
+		vimCmd.Stdout = os.Stdout
+		if err = vimCmd.Run(); err != nil {
+			//FIXME: Don't abort immediately, but try to edit remaining files?
+			return false, err
+		}
+
+		// Get modification time after closing the editor
+		info, err = os.Stat(plugconfPath)
+		if err != nil {
+			return false, err
+		}
+		mTimeAfter := info.ModTime()
+
+		// A change was made if the modification time was updated
+		changeWasMade = changeWasMade || mTimeAfter.After(mTimeBefore)
+
+		// Remove repository from lock.json
+		err = lockJSON.Repos.RemoveAllReposPath(reposPath)
+		err2 := lockJSON.Profiles.RemoveAllReposPath(reposPath)
+		if err == nil || err2 == nil {
+			// ignore?
+		}
+	}
+
+	// Write to lock.json
+	if err = lockJSON.Write(); err != nil {
+		return changeWasMade, err
+	}
+	return changeWasMade, nil
+}
+
+func (cmd *editCmd) parseArgs(args []string) (pathutil.ReposPathList, error) {
+	fs := cmd.FlagSet()
+	fs.Parse(args)
+	if cmd.helped {
+		return nil, ErrShowedHelp
+	}
+
+	if len(fs.Args()) == 0 {
+		fs.Usage()
+		return nil, errors.New("repository was not given")
+	}
+
+	// Normalize repos path
+	reposPathList := make(pathutil.ReposPathList, 0, len(fs.Args()))
+	for _, arg := range fs.Args() {
+		reposPath, err := pathutil.NormalizeRepos(arg)
+		if err != nil {
+			return nil, err
+		}
+		reposPathList = append(reposPathList, reposPath)
+	}
+
+	return reposPathList, nil
+}
+
+func (cmd *editCmd) identifyEditor(cfg *config.Config) (string, error) {
+	var editors []string
```

Comment:
please change to

```suggestion
	editors := make([]string, 0, 10)
```

to avoid unnecessary allocation.

Reply:
I have changed it to 
`editors := make([]string, 4, 6)`
since we will add at least 4 and add max 6 items to it. Wouldn't that be better? I am new to go and not very used to such details...

Reply:
The 2nd argument of `make()` is the default number of slice.
`make([]string, 4, 6)` creates a slice of 4 empty elements (in this case, empty strings).
https://play.golang.org/p/jmH0rHiS32f

Maybe your change does not work as you expect.

Reply:
You are right. A have changed it to `make([]string, 0, 6)` now.